File: Cpp/1423_Maximum_Points_You_Can_Obtain_from_Cards.cpp
class Solution {
public:
    int maxScore(vector<int>& cardPoints, int k) {
        int n = cardPoints.size();
        if(n <= k) return accumulate(cardPoints.begin(), cardPoints.end(), 0);
        
        int sum = 0;
        for(int i = 0; i<k; ++i) {
            sum += cardPoints[i];
        }
        
        int result = sum;
        for(int i=0; i<k; ++i) {
            sum -= cardPoints[k-i-1];
            sum += cardPoints[n-i-1];
            result = max(result, sum);
        }
        
        return result;
    }
};

File: Kotlin/1423_Maximum_Points_You_Can_Obtain_from_Cards.kt
class Solution {
    fun maxScore(cardPoints: IntArray, k: Int): Int {
        val n = cardPoints.size
        if(n <= k) return cardPoints.sum()
        
        var sum = cardPoints.filterIndexed { index, _ -> index < k }.sum()
        var result = sum
        
        for(i in 0 until k) {
            sum -= cardPoints[k - i - 1]
            sum += cardPoints[n - i - 1]
            result = maxOf(result, sum)
        }
        
        return result
    }
}